### DISCUSSION TITLE
How to generate a token ID in Kotlin/Java for production use?

### DISCUSSION THREAD

**Discussion by DoSzero** [GetStream/stream-chat-android#1980] [General]
**How to generate a token ID in Kotlin / Java  for production use?**
I didn't get provider in the website documentation .  (https://getstream.io/chat/docs/react/tokens_and_authentication/?language=java) Thanks for your project , it helped a lot 👍🏻

**Comment by leandroBorgesFerreira** [comment]
Hello @DoSzero. This must be generated in your backend. Than you can implement your `TokenProvider` using an end point of your backend of any other way that you prefer.

**Comment by rahulsain** [comment]
how can we use token in android without having a backend?
devToken or hardcoded will work or not?
